src/fs/filesystem.ts

```diff
@@ -25,16 +27,37 @@ export class FileSystem {
   publicTree: HeaderTree
   prettyTree: PublicTreeBare
   privateTree: HeaderTree
+
   syncHooks: Array<SyncHook>
+  syncWhenOnline: CID | null
+
 
   constructor({ root, publicTree, prettyTree, privateTree }: ConstructorParams) {
     this.root = root
     this.publicTree = publicTree
     this.prettyTree = prettyTree
     this.privateTree = privateTree
+
     this.syncHooks = []
+    this.syncWhenOnline = null
+
+    // Update the user's data root when making changes
+    const syncHook = throttle(5000, cid => {
```

Comment:
dig the throttle :+1: 
but could we wind up in a situation where a user fires off a number of updates in quick succession, and only the first gets synced?

Reply:
I don't believe so, it should fire the first call and the last call, and there will always be 5 seconds in between (so last call will be delayed when necessary). See readme for more info: https://www.npmjs.com/package/throttle-debounce#throttle

Reply:
ah got it got it, that's what the `noTrailing` param that it mentions would do. cool this looks fine then :+1: